be1-go/db/sqlite.go

```diff
@@ -0,0 +1,199 @@
+package db
+
+import (
+	"database/sql"
+	"encoding/base64"
+	"encoding/json"
+	"log"
+	"math"
+	"os"
+	"student20_pop/message"
+
+	_ "github.com/mattn/go-sqlite3"
+	"golang.org/x/xerrors"
+)
+
+type sqlite struct {
+	conn *sql.DB
+}
+
+var (
+	DDL = `
+	CREATE TABLE messages (
+		channel_id text, message_id text, timestamp integer, message text,
```

Comment:
I would also create an index on message_id , but that might be something for the students to implement as part of the improved  catch-up logic :)